coral/src/app/features/requests/components/topics/TopicRequests.tsx

```diff
@@ -4,20 +4,38 @@ import { getTopicRequests } from "src/domain/topic/topic-api";
 import { TopicRequestsTable } from "src/app/features/requests/components/topics/components/TopicRequestsTable";
 import { useSearchParams } from "react-router-dom";
 import TopicFilter from "src/app/features/components/table-filters/TopicFilter";
+import { Pagination } from "src/app/components/Pagination";
 
 function TopicRequests() {
-  const [searchParams] = useSearchParams();
+  const [searchParams, setSearchParams] = useSearchParams();
   const currentTopic = searchParams.get("topic") ?? "";
+  const currentPage = searchParams.get("page")
+    ? Number(searchParams.get("page"))
+    : 1;
 
   const { data, isLoading, isError, error } = useQuery({
-    queryKey: ["topicRequests", currentTopic],
+    queryKey: ["topicRequests", currentTopic, currentPage],
     queryFn: () =>
       getTopicRequests({
-        pageNo: "1",
+        pageNo: String(currentPage),
```

Comment:
I think we also need to have the 
```
    keepPreviousData: true,
```
option in this query so we don't flash the `SkeletonTable` on page change (keeping previous table state instead).